### DISCUSSION TITLE
Go: A question about uploading objects

### DISCUSSION THREAD

**Discussion by phllpmcphrsn** [minio/minio#16067] [Q&A] [answered]
Does a successful response from uploading an object mean that it was fully uploaded or does it simply mean that MinIO got the request and has started to uploaded the object? I'm using AWS' s3 SDK for Go (what I was told to use - don't judge me), and I have a situation where it's seems to be returning successfully but the object isn't fully uploaded by the time I need to pull from it. The SDK states that 
>  Amazon S3 never adds partial objects; if you receive a success response, Amazon S3 added the entire object to the bucket.

However, I'm not sure if MinIO abides by that as well

**Comment by harshavardhana** [comment, marked as answer]
> > Amazon S3 never adds partial objects; if you receive a success response, Amazon S3 added the entire object to the bucket.

MinIO is strictly consistent it means if you got 200 OK it means the object has been successfully persisted to disk.
